react-carousel-component/justin-app/src/Carousel.js

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from 'react';
+import { AiOutlineLeft, AiOutlineRight } from 'react-icons/ai';
+import { RxDotFilled, RxDot } from 'react-icons/rx';
+import './Carousel.css';
+
+export default function Carousel({ items }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setCurrentIndex((currentIndex + 1) % items.length);
+    }, 3000);
+    return () => clearInterval(intervalId);
+  }, [currentIndex, items.length]);
+
+  function handleClickNext() {
```

Comment:
FYI, you could call this in the Effect if you wrapped it in `useCallback`, though we haven't studied that yet.